corehq/apps/data_dictionary/util.py

```diff
@@ -192,9 +192,13 @@ def save_case_property(name, case_type, domain=None, data_type=None,
         prop.data_type = data_type
     if description is not None:
         prop.description = description
-    if group:
+    if group is not None:
         prop.group = group
-        prop.group_obj, created = CasePropertyGroup.objects.get_or_create(name=group, case_type=prop.case_type)
+        # Allow properties to have no group
+        if group:
```

Comment:
What's going on here?  When would `group` be `not None` but also `Falsy`?  Are you expecting callers of this function to sometimes not pass in `group` and to sometimes pass in `group=""`, with the latter indicating that the group should be deleted?

If so, someone calling this function with `group=None` might be confused to find that the group is not deleted.  An alternative you might be interested in is using `...` as the default value to identify unset arguments, as demonstrated in this recent PR
https://github.com/dimagi/commcare-hq/pull/32652/

Reply:
I have made the code simpler for this. Saving the group on truthy value and deleting the group for falsy value. This way on both empty string "" and None the group will be deleted. [fde6396](https://github.com/dimagi/commcare-hq/pull/32904/commits/fde6396ad1729d694c35b25cc1d94d77e635e728)

Reply:
That makes sense, and I like the simpler code, though I think that might still run into some issues.

I see this call to the function, which doesn't provide a `group`:
https://github.com/dimagi/commcare-hq/blob/d7661872932367ce9e19cda52877ad2b914c82b1/corehq/apps/data_dictionary/views.py#L206-L206

Also, from the way the function is written, it looked like it was meant to support partial updates without deleting fields that aren't provided (each property is only set if it's not falsy).  I'm not sure if the browser sends the full serialized version of the property each time or if it only sends properties that have been updated, but that could impact this as well.

Actually, looking at this more closely the way this is written (prior to your changes, I mean) is troublesome due to this confusion.  For instance, I just discovered that you cannot deselect a datatype.  If you set the dropdown to "Select a data type", hit save, then reload the page, it will revert to the previous data type.

Reply:
Thanks for the info @esoergel. Using `...` for unset is the best approach for fixing this bug as you suggested. The same fix can also work for `data_type` also. 
[82d69fa](https://github.com/dimagi/commcare-hq/pull/32904/commits/82d69fa0c9c0cc83db71b154bfffe1070fb9d51f)
[e34c107](https://github.com/dimagi/commcare-hq/pull/32904/commits/e34c107efc9ad3f150502dd095b0e2d90e8753c8)

Reply:
Nice, thanks, that does look like it should fix those two.

Would you consider switching the other properties to use `...`?  I worry about it being confusing to have two different types of unset params for the same set of args.

One thing worth sanity checking though is whether the front-end ever sends partially specified properties.  This block of code (below) uses `property.get('name')` instead of `property['name']` as if those fields might sometimes not be provided, and if so, switching the default from `None` to `...` could have unintended consequences.
https://github.com/dimagi/commcare-hq/blob/0b27503b59351347d8e3531746fed33c500e4841/corehq/apps/data_dictionary/views.py#L157-L165

Reply:
> Would you consider switching the other properties to use ...? I worry about it being confusing to have two different types of unset params for the same set of args.

I was thinking to only have the `...` for unset for properties which have falsy states (""), which are group and data_type field here. Rest of the fields are working as intended.

>One thing worth sanity checking though is whether the front-end ever sends partially specified properties. This block of code (below) uses property.get('name') instead of property['name'] as if those fields might sometimes not be provided, and if so, switching the default from None to ... could have unintended consequences.

I looked at the API payload and only the data_type field is sent as `None` when no data_type is selected. group is sent as `""` (empty string) for `No Group`.

Reply:
Hey @esoergel @pxwxnvermx 

Can I ask us to go back to the original fix that was made for this issue?

https://github.com/dimagi/commcare-hq/pull/32904/commits/5adc1e26c127489a03817b8c0e296e56e5db93e0#diff-35bed86d7b8b1e2a0884ce4d860754dbc3d2787834b3001019e239da00655da9R194-R195

For few reasons:
1. firstly, that looks like a safe change, which is more predictable in terms of what it changes. We know users can pass an empty string from UI or from excel upload, in which case group should be removed, so we handled that. The function call where group is not set, there is no change to it's expectations. The other calls from UI and excel upload would always pass some value, empty string or `None`. Adding `...` adds nested Ifs which makes this function more complex and harder to predict the impact.
2. group is a string value, so thinking of "None" to be considered as equivalent to "No action" seems okay to me here. And that if someone wants group to be removed, an empty string would be the right thing to pass.
3. The only place where group is not passed, its expectations where left unchanged anyway by the original fix (as shared in point 1 above), so adding `...` isn't changing anything about the functionality.
4. If we are to improve the functionality of this method, by making most args `...`, let's take that in a separate PR. I am bit unsure of what other impact it could have and I am not comfortable including that in this PR. 

Open to feedback on any of those points above. I hope I was able to follow the conversation so far correctly.
Though keeping the scope of this PR to what it originally intended, would definitely be good.

Though, I think using `None` as a way to say, `No Change` for string properties, is probably okay. The change that Norman made was to remove something that was replacing `...` i.e `CaseBlock.undefined`, so it did make sense there.

Reply:
Hey @esoergel 
any thoughts on this?

Reply:
Thanks for re-pinging, @mkangia yes, I think that should be okay as you described - this is getting off-track.

Reply:
thanks @esoergel 

@pxwxnvermx i think we are good to just remove the commits that are not needed anymore

Reply:
Thanks @esoergel @mkangia 

I have removed the commits that were not needed and added a fix for allowing to remove the data_type. [13a9322](https://github.com/dimagi/commcare-hq/pull/32904/commits/13a9322fd273e9c4efdfc640169abf082ad0e047)